### PR TITLE
Use Traject instead of SolrMarc

### PR DESCRIPTION
**Pull request by cbeer**

### CODE REVIEW

File: spec/test_app_templates/lib/generators/test_app_generator.rb [hunk 1]
```diff
@@ -18,7 +18,8 @@ def remove_index
   def run_blacklight_generator
     say_status("warning", "GENERATING BL", :yellow) 
     gem 'blacklight'
-
+    gem "traject", path:'/Users/benjamin/github/barmintor/traject'
```
Comment (cbeer) [inline spec/test_app_templates/lib/generators/test_app_generator.rb]: Want to fix this?

File: lib/blacklight/marc/indexer/formats.rb [hunk 2]
```diff
@@ -0,0 +1,185 @@
+module Blacklight::Marc::Indexer::Formats
+  module FormatMap
+    def self.map007(v, vals)
+      field007hasC = false
+      v = v.upcase
+      if v.start_with? 'A'
+        vals << (v == 'AD') ? 'Atlas' : 'Map'
+      elsif v.start_with? 'C'
+        if (v == "CA")
```
Comment (cbeer) [inline lib/blacklight/marc/indexer/formats.rb]: Can this be `case`/`when`?

File: lib/generators/blacklight_marc/templates/app/models/marc_indexer.rb [hunk 3]
```diff
@@ -0,0 +1,150 @@
+$:.unshift './config/SolrMarc'
```
Comment (cbeer) [inline lib/generators/blacklight_marc/templates/app/models/marc_indexer.rb]: What's `config/SolrMarc` used for?

File: lib/blacklight/marc/indexer.rb [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+  def initialize
+    super
+
+    settings do
+      provide "reader_class_name", "Traject::MarcReader"
+      provide "marc_source.type", "binary"
+      if c = Blacklight.solr_config
+        provide "solr.url", c[:url]
+      end
```
Comment (cbeer) [inline lib/blacklight/marc/indexer.rb]: Should this be `elsif`?

File: lib/blacklight/marc/indexer.rb [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+  ATOZ = ('a'..'z').to_a.join('')
+  ATOU = ('a'..'u').to_a.join('')
+
+  def initialize
+    super
+
+    settings do
+      provide "reader_class_name", "Traject::MarcReader"
+      provide "marc_source.type", "binary"
```
Comment (cbeer) [inline lib/blacklight/marc/indexer.rb]: Does this work with both marc21 and marc-xml?

**Comment by cbeer** [comment]
@barmintor++

File: lib/generators/blacklight_marc/templates/app/models/marc_indexer.rb (hunk 3 above)
Comment (barmintor) [inline lib/generators/blacklight_marc/templates/app/models/marc_indexer.rb]: Traject is (at least for now, afaik) insistent on looking the property files up on the classpath, and that's the legacy directory.

File: lib/blacklight/marc/indexer.rb (hunk 5 above)
Comment (barmintor) [inline lib/blacklight/marc/indexer.rb]: as far as I know, the ruby-marc reader dispatches to an appropriate parser. But I'll look for some xml to test against.

File: lib/blacklight/marc/indexer/formats.rb (hunk 2 above)
Comment (barmintor) [inline lib/blacklight/marc/indexer/formats.rb]: Yeah, this was me porting the Java too directly.

File: spec/test_app_templates/lib/generators/test_app_generator.rb (hunk 1 above)
Comment (barmintor) [inline spec/test_app_templates/lib/generators/test_app_generator.rb]: GAAAAH

File: lib/blacklight/marc/indexer.rb (hunk 4 above)
Comment (barmintor) [inline lib/blacklight/marc/indexer.rb]: I'm not sure. What do you think? Should the env override BL?

**Comment by jcoyne** [comment]
@barmintor Nice! :star: Ready to merge now?

File: README.md [hunk 6]
```diff
@@ -18,6 +18,23 @@ After running the blacklight generator, run the blacklight_marc generator:
...
+    # and define fields as normal with the mixed-in methods
+    to_field 'format', get_format
+    to_field 'dewey100', dewey_facets(base: :hundreds)
+    to_field 'dewey010', dewey_facets(base: :tens)
+    to_field 'dewey_text', dewey_text
+  end
+end
+```
 ## Features
 * Rake task `solr:marc:index` to import .mrc files using SolrMarc
```
Comment (jcoyne) [inline README.md]: Does this line need to be updated?

**Comment by cbeer** [comment]
@barmintor what's left? I took the branch for a spin the other day and it seemed to work great.

**Comment by barmintor** [comment]
I had been trying to put together a faster fork of traject- I think it works fine, I was just concerned about speed.

**Comment by barmintor** [comment]
Given that there's no pinned version of Traject here, even that's not a great reason. If y'all think it looks good, go for it.

**Comment by barmintor** [comment]
@cbeer @jcoyne I would love to see this merged for BL 6. Let me see if I can find some time to run tests again with the latest release to make sure nothing has broken.

**Comment by cbeer** [comment]
:+1:

**Comment by jcoyne** [comment]
@barmintor find time yet?

**Comment by barmintor** [comment]
Just rebased, running ci now.

**Comment by barmintor** [comment]
I need to get the generator and ci indexing working again.

**Comment by barmintor** [comment]
@jcoyne assuming CI passes, I assume you'd like this to be less than 17 commits.

**Comment by jcoyne** [comment]
@barmintor Yeah, unless there's some benefit to isolating these commits, squash 'em.

File: Gemfile [hunk 7]
```diff
@@ -33,3 +33,5 @@ else
   end
 end
 # END ENGINE_CART BLOCK
+
+gem 'transpec'
```
Comment (jcoyne) [inline Gemfile]: Do we need to leave this in the Gemfile?

Comment (barmintor) [inline Gemfile]: Apparently not. Deleting, amending, testing.

**Comment by barmintor** [comment]
Any other notes before I switch gears? @jcoyne @cbeer ?

**Comment by jcoyne** [comment]
@barmintor I don't understand why you are preserving `config/SolrMarc`? Can you explain it to me?

**Comment by barmintor** [comment]
@jcoyne just to minimize renaming translation map files at this point; traject used to be fussy about it, but I guess it's not anymore?

**Comment by barmintor** [comment]
@jcoyne TBH I'm pretty far out of context about this (Feb 2015 was a long time ago).

**Comment by barmintor** [comment]
@jcoyne ok, so traject wants the maps on the load path. I feel like this stuff should be in config, so I have to add *something* to the load path. If I change the path addition to just be './config', and move one of the generator's property files to config/translation_maps/, it works. What do you think?

**Comment by jcoyne** [comment]
@barmintor is the line in README.md still valid? 
> Rake task `solr:marc:index` to import .mrc files using SolrMarc

**Comment by barmintor** [comment]
@jcoyne sorry didn't understand what you meant earlier. Fixed, pushed, etc.

**Comment by barmintor** [comment]
@jcoyne the branch has been amended to move all the existing translation maps to config/translation_maps, delete config/SolrMarc, and edit the generator's export appropriately.

**Pull request merged**